clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/MoveExecutor.scala

```diff
@@ -124,7 +129,14 @@ class MoveExecutor[TI <: TransferIndex](moveCommand: MoveCommand[TI])
       )
     } else if (movesToPerform.isEmpty) {
       logger.info(s"Nothing to move; all files already exist at $destination")
-      Future.successful(None)
+
+      // Although we don't need to actually perform the move, we may need to perform custom delivery or other steps.
```

Comment:
This block might be a bad idea. I'm not sure of a better solution. Open to suggestions.